sys/exec.c

```diff
@@ -86,13 +87,16 @@ int do_exec(const exec_args_t *args) {
     return -ENOEXEC;
   }
 
-  /* TODO: Get current process description structure */
+  thread_t *td = thread_self();
 
-  /* The current vmap should be taken from the process description! */
-  vm_map_t *old_vmap = get_user_vm_map();
-  /* We may not destroy the current vm map, because exec can still
-   * fail, and in that case we must be able to return to the
-   * original address space
+  /* Please don't switch to other threads while we prepare the address
```

Comment:
In fact, there's no distinction between kernel and user threads. Both are the same – at any given moment a thread can execute either kernel-space or user-space code (provided that a thread can safely enter user-mode). That's all.

According to classic OS textbooks our threads are Kernel Level Threads in 1-to-1 model.